lib/test_kfence.c

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Test cases for SL[AOU]B/page initialization at alloc/free time.
```

Comment:
I'm currently working on KCSAN tests using KUnit+Torture frameworks. We probably need to change this to use those eventually as well. But for now this is probably fine.

I'm happy to convert once I figured out the KCSAN tests.

Reply:
I just wrote a quick-and-dirty test to debug error reporting, but I agree KUnit is a must.

Reply:
(also the above snippet contains an irrelevant comment, I'll fix that)